source/extensions/tracers/opentelemetry/resource_detectors/dynatrace/dynatrace_metadata_file_reader.cc

```diff
@@ -0,0 +1,56 @@
+#include "source/extensions/tracers/opentelemetry/resource_detectors/dynatrace/dynatrace_metadata_file_reader.h"
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "envoy/common/exception.h"
+
+#include "source/common/common/logger.h"
+
+#include "absl/strings/str_cat.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace OpenTelemetry {
+
+namespace {
+
+bool isIndirectionFile(const std::string& file_name) {
+  return absl::StartsWith(file_name, "dt_metadata_");
+}
+
+const std::string readFile(const std::string& file_name) {
+  if (file_name.empty()) {
+    return "";
+  }
+
+  std::ifstream file(file_name);
+  if (file.fail()) {
+    throw EnvoyException(absl::StrCat("Unable to read Dynatrace enrichment file: ", file_name));
+  }
+
+  std::stringstream file_string;
+  file_string << file.rdbuf();
+
+  return file_string.str();
+}
+
+} // namespace
+
+std::string DynatraceMetadataFileReaderImpl::readEnrichmentFile(const std::string& file_name) {
+  bool indirectionFile = isIndirectionFile(file_name);
+  if (indirectionFile) {
+    std::string indirection_file = readFile(file_name);
```

Comment:
Reading file system is a blocking I/O in a data path, so that might be a performance penalty?

Reply:
If I understood correctly from my others PRs, this is called during Envoy start up, so it's the "configuration path". This is called only once, when starting up the Tracer. So I think it's not a problem and the files we read are controlled by Dynatrace and are always very small, meaning there won't be a lot of data inside.